google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/StreamingSubscriberConnection.java

```diff
@@ -541,6 +542,7 @@ public void onFailure(Throwable t) {
           Map<String, String> metadataMap = getMetadataMapFromThrowable(t);
           ackRequestDataList.forEach(
               ackRequestData -> {
+                messageDispatcher.notifyAckFailed(ackRequestData);
```

Comment:
We can't notify ack failed right here because we may retry below. We should only call this method if we set a response on ackRequestData below

Reply:
Done, added notifyAckSuccess() below as well.